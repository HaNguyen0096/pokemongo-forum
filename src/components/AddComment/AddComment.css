
.addComment{
  padding: 10px;
}

.postCommentBtn{
  background-color: rgb(23,156,106);
  color: white;
  padding: 16px 20px;
  border: none;
  cursor: pointer;
  width: 235px;
  margin: 10px;
  margin-top: -10px;
  opacity: 0.9;
}

