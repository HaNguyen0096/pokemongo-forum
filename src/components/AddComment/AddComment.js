import React, { Component } from 'react'
import MainContext from '../../contexts/TopicContext'
import ApiService from '../../services/api-service'
import { Button, Textarea } from '../Utils/Utils'

export default class AddComment extends Component {
  
  static contextType = MainContext

  handleSubmit = ev => {
    ev.preventDefault()
    const { thread } = this.props
    console.log(thread)
    const { text } = ev.target

    ApiService.postComment(thread.thread_id, text.value)
      .then(this.context.addComment)
      .then(() => {
        text.value = ''
      })
      .catch(this.context.setError)
  }

  render() {
    return (
      <form
        className='Comment'
        onSubmit={this.handleSubmit}
      >
        <div className='text'>
          <Textarea
            required
            aria-label='Type a comment...'
            name='text'
            id='text'
            cols='30'
            rows='3'
            placeholder='Type a comment..'>
          </Textarea>
        </div>

        <Button type='submit'>
          Post comment
        </Button>
      </form>
    )
  }
}
