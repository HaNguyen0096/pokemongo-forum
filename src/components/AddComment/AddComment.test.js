import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddComment from './AddComment'

describe('AddComment component', () => {

  it('renders AddComment without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                      <AddComment />
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI for AddComment as expected', () => {
    const tree = renderer
      .create(<BrowserRouter>
                <AddComment />
              </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

  it('renders AddComment by default', () => {
    const wrapper = shallow(<AddComment  />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

})