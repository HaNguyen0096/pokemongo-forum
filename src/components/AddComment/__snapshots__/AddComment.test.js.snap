// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddComment component renders AddComment by default 1`] = `
<form
  className="Comment"
  onSubmit={[Function]}
>
  <div
    className="addComment"
  >
    <Textarea
      aria-label="Type a comment..."
      cols="30"
      id="text"
      name="text"
      placeholder="Type a comment.."
      required={true}
      rows="3"
    />
  </div>
  <Button
    className="postCommentBtn"
    type="submit"
  >
    Post comment
  </Button>
</form>
`;

exports[`AddComment component renders the UI for AddComment as expected 1`] = `
<form
  className="Comment"
  onSubmit={[Function]}
>
  <div
    className="addComment"
  >
    <textarea
      aria-label="Type a comment..."
      className="Textarea "
      cols="30"
      id="text"
      name="text"
      placeholder="Type a comment.."
      required={true}
      rows="3"
    />
  </div>
  <button
    className="Button postCommentBtn"
    type="submit"
  >
    Post comment
  </button>
</form>
`;
